command.go

```diff
@@ -923,9 +927,12 @@ func (c *Command) execute(a []string) (err error) {
 	}
 
 	if !c.Runnable() {
-		return flag.ErrHelp
+		if EnableErrorOnUnknownSubcommand {
+			return ErrUnknownSubcommand
```

Comment:
Let's only return the error if there is at least one arg.
This will allow to get the usage printed without an error when running a valid command that doesn't have a Run.  This is something users do a lot.  For example running `helm repo` should print the usage without showing an error.

You should be able to move up the following lines to get the number of args:
```
argWoFlags := c.Flags().Args()
	if c.DisableFlagParsing {
		argWoFlags = a
	}
```